docs/flow.rst

```diff
@@ -0,0 +1,235 @@
+===================
+Resolwe Flow Design
+===================
+
+The Resolwe Flow dataflow engine comprises the execution framework and the
+management layers which make up the internal data model and facilitate
+dependency resolution, permissions enforcement and process filtering.
+
+Overview
+========
+
+Execution Framework
+-------------------
+
+Flow consists of a number of active services which need to be running before job
+execution can proceed.
+
+The core message transport and coordination facility, as currently used, is
+`Redis`_. It serves as the central status hub for keeping track of shared
+dynamic information about the manager instances, and as a contact point for
```

Comment:
Following this strictly would lead to content layout that would make no sense. Certain things have to be referenced before they're defined. I've reworded the text to get rid of the references but otherwise kept it the same.